### PR TITLE
Bug 2040150: Make changes to HPCS KMS component

### PR DESCRIPTION
**Pull request by SanjalKatiyar**
- Made changes to the keys used in the ConfigMaps.
- Ceph-CSI will also follow CAPITAL_CASE format for keys for HPCS.
- Removed "Advanced settings" modal for HPCS.
- BASE_URL is now a required field.
- TOKEN_URL is still optional.

Screenshot from 2022-01-12 23-47-58

Screenshot from 2022-01-12 23-48-18

### CODE REVIEW

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx [hunk 1]
```diff
@@ -135,14 +135,14 @@ const generateHpcsSecret = (kms: HpcsConfig) => ({
   apiVersion: SecretModel.apiVersion,
   kind: SecretModel.kind,
   metadata: {
-    name: `ibm-kms-key-${Math.random()
+    name: `ibm-kp-kms-${Math.random()
       .toString(36)
       .substring(7)}`,
```
Comment (bipuladh) [inline frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx:140]: Nit: Use `getRandomChars` from @console/shared.

**Comment by yuvalgalanti** [comment]
looks great

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/hpcs-config.tsx [hunk 2]
```diff
@@ -87,7 +61,7 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({ state, dispatch, cl
       >
         <TextInput
           value={kms.name?.value}
-          onChange={setServiceName}
+          onChange={setRequiredParams('name')}
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/kms-config/hpcs-config.tsx:64]: You converted into a single generic function that's good, but it introduced the risk of maintaining key names in multiple places.

A generic function is feasible only when enough constants and enum are shared between callers. Either you need to convert all HPCS keys into enum/constants and reuse them in this function call and the object declaration (or)  the old approach i.e individual functions for each key.

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx [hunk 3]
```diff
@@ -248,15 +248,15 @@ const createCsiHpcsResources = (
...
+    IBM_KP_SECRET_NAME: secretName || keySecret.metadata.name,
+    IBM_KP_BASE_URL: kms.baseUrl.value,
+    IBM_KP_TOKEN_URL: kms.tokenUrl,
   };
-  const csiConfigObj: ConfigMapKind = generateCsiKmsConfigMap(kms.name.value, csiConfigData);
+  const csiConfigObj: ConfigMapKind = generateCsiKmsConfigMap(kms.name.value, csiConfigData, false);
 
   if (update) {
-    const cmPatch = [generateConfigMapPatch(kms.name.value, csiConfigData)];
+    const cmPatch = [generateConfigMapPatch(kms.name.value, csiConfigData, false)];
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/kms-config/utils.tsx:259]: Is CSI is not supporting Camelcase for HPSC?

File: frontend/packages/ceph-storage-plugin/src/components/kms-config/hpcs-config.tsx [hunk 4]
```diff
@@ -28,48 +27,23 @@ export const HpcsConfigure: React.FC<KMSConfigureProps> = ({ state, dispatch, cl
...
-      payload: kmsObj,
-    });
-  };
-
-  const setApiKey = (apiKey: string) => {
-    kmsObj.apiKey.value = apiKey;
-    kmsObj.apiKey.valid = apiKey !== '';
+  const setRequiredParams = (param: string) => (value: string) => {
+    kmsObj[param].value = value;
+    kmsObj[param].valid = value !== '';
```
Comment (GowthamShanmugam) [inline frontend/packages/ceph-storage-plugin/src/components/kms-config/hpcs-config.tsx:32]:
```suggestion
  const setRequiredParams = (param: string, validate: boolean = true) => (value: string) =>{
     if validate{
         kmsObj[param].valid = value !== '';
     }
     kmsObj[param].value = value;
     dispatch({
        type: 'securityAndNetwork/setHpcs',
        payload: kmsObj,
    });
  }
```

**Pull request merged**